src/translate-adapter.ts

```diff
@@ -0,0 +1,455 @@
+	.options({
+		ioPackage: {
+			type: "string",
+			alias: "p",
+			default: "./io-package.json",
+			description: "Path to the io-package.json file",
+		},
+		admin: {
+			type: "string",
+			alias: "a",
+			default: "./admin",
+			description: "Path to the admin directory",
+		},
+		words: {
+			type: "string",
+			alias: "w",
+			description: "Path to the words.js file",
+		},
+		base: {
+			type: "string",
+			alias: "b",
+			array: true,
+			description:
+				"Path to the english i18n file, multiple files are possible",
+		},
+	})
+	.middleware(setDirectories)
+	.wrap(Math.min(100, parser.terminalWidth()))
+	.help().argv;
+
+/********************************** Helpers ***********************************/
+
+const _languages = {
+	en: {},
+	de: {},
+	ru: {},
+	pt: {},
+	nl: {},
+	fr: {},
+	it: {},
+	es: {},
+	pl: {},
+	"zh-cn": {},
+};
+
+type Language = keyof typeof _languages;
+
+function getLanguages(): Language[] {
+	return Object.keys(_languages) as Language[];
+}
+
+function createEmptyLangObject<T>(createDefault: () => T): Record<Language, T> {
+	return getLanguages().reduce(
+		(obj, curr) => ({ ...obj, [curr]: createDefault() }),
+		{} as Record<Language, T>,
+	);
+}
+
+/**
+ * Creates a regexp pattern for an english base file name.
+ * It matches file names and allows to find/replace the language code
+ */
+function createFilePattern(baseFile: string): RegExp {
+	if (!baseFile.match(/\Wen\W/)) {
+		throw new Error("Base file must be an English JSON file");
+	}
+	return new RegExp(
+		`^(${escapeRegExp(baseFile).replace(
+			/(?<=\W)en(?=\W)/,
+			")([a-z-]+)(",
+		)})$`,
+		"i",
+	);
+}
+
+async function findAllLanguageFiles(baseFile: string): Promise<string[]> {
+	const filePattern = createFilePattern(baseFile);
+	const allJsonFiles = await promisify(glob)(
+		path.join(admin, "**", "*.json"),
+		{ realpath: true },
+	);
+	const languages = getLanguages();
+	return allJsonFiles.filter((file) => {
+		const match = file.match(filePattern);
+		if (!match) {
+			return false;
+		}
+		const lang = match[2] as Language;
+		return languages.includes(lang);
+	});
+}
+
+function die(message: string): never {
+	console.error(red(message));
+	process.exit(1);
+}
+
+function interceptErrors(func: () => Promise<void>): () => Promise<void> {
+	return async () => {
+		try {
+			await func();
+		} catch (error) {
+			die(error.stack || error);
+		}
+	};
+}
+
+/******************************** Middlewares *********************************/
+
+async function setDirectories(options: {
+	ioPackage: string;
+	admin: string;
+	words?: string;
+	base?: string[];
+}): Promise<void> {
+	// io-package.json
+	ioPackage = path.resolve(options.ioPackage);
+	if (!existsSync(ioPackage) || !(await stat(ioPackage)).isFile()) {
+		return die(`Couldn't find file ${ioPackage}`);
+	}
+
+	// admin directory
+	admin = path.resolve(options.admin);
+	if (!existsSync(admin) || !(await stat(admin)).isDirectory()) {
+		return die(`Couldn't find directory ${admin}`);
```

Comment:
I **think** TypeScript understands that never-returning functions don't return. Shouldn't need the `return` here but it doesn't hurt either.

Reply:
Yes, it would work. The `return` was partially also to make the code more readable (it is immediately clear that execution doesn't continue).